actions/google-drive/actions.py

```diff
@@ -122,54 +337,101 @@ def get_files_by_query(
         list[
             Literal[
                 "https://www.googleapis.com/auth/drive.readonly",
-                "https://www.googleapis.com/auth/drive.metadata.readonly",
+                "https://www.googleapis.com/auth/drive.metadata.readonly"
             ]
         ],
     ],
     query: str,
     search_all_drives: bool = False,
+    basic_info_only: bool = False,
+    save_result_as_csv: Union[bool, str] = False,
 ) -> Response[FileList]:
     """Get all files from Google Drive that match the given query.
 
     Args:
         google_credentials: JSON containing Google OAuth2 credentials.
         query: Google Drive API V3 query string for search files in the format query_term operator values.
+            When using 'in parents' operator, you can use either the parent folder's ID or name.
+            Example: "name = 'Report' and 'My Folder' in parents"
+            When search_all_drives is True, parent folder names will be searched in shared drives first.
         search_all_drives: Whether to search both My Drives and all shared drives.
             Default is set to false for better performance and will include both owned files and those that have been
             shared with the user.
-
+        basic_info_only: Whether to return only the basic information of the files
+        save_result_as_csv: If True, saves results to 'query_result.csv'. If a string is provided,
+            uses that as the filename to save the CSV results.
 
     Returns:
         A list of files or an error message if no files were found.
     """
     service = _build_service(google_credentials)
 
-    list_args = {
-        "supportsAllDrives": True,
-        "includeItemsFromAllDrives": True,
-        "q": query,
-        "fields": "*",
-    }
-
-    if search_all_drives:
-        list_args["corpora"] = "allDrives"
-        list_args["spaces"] = "drive"
-
-    response = service.files().list(**list_args).execute()
-    service.close()
-
-    files = FileList(files=response.get("files", []))
-    if not files:
-        return Response(error=f"No files were found for the query: {query}")
-
-    return Response(result=files)
+    try:
+        # First try to resolve any parent folder names
+        processed_query = _preprocess_query(service, query, search_all_drives)
+
+        list_args = {
+            "supportsAllDrives": True,
+            "includeItemsFromAllDrives": True,
+            "q": processed_query,
+            "fields": "*",
```

Comment:
Is there any chance of this creating long running actions?
Not sure if one of the fields is `content`...